pegen/c_generator.py

```diff
@@ -396,6 +410,11 @@ def visit_Rule(self, node: Rule) -> None:
             self._set_up_rule_memoization(node, result_type)
 
         self.print("{")
+        if node.name.startswith("start"):
+            with self.indent():
+                self.print(
+                    f"init_keywords(p, &keywords, {len(self.callmakervisitor.keyword_cache)});"
+                )
```

Comment:
Its unfortunate that we have to do this in a specific rule, since in the future we need to be able to use different start rules for the same grammar (e.g. single_input, file_input, eval_input in the pgen Grammar file). But the only way I can think of is to pass the list of keywords to the run_parser* functions. What do you think?

Reply:
Hadn't thought of this! It seems like a better approach for sure. Should I update this PR?

Reply:
Go for it.